src/connectors/tcp_connector/test/tcp_connector_test.cc

```diff
@@ -135,14 +135,14 @@ ssize_t recv (int, void *buf, size_t n, int)
         return 0;
 }
 
-#ifdef __FreeBSD__
-int socket (int, int, int) { return s_socket_return; }
-int bind (int, const struct sockaddr*, socklen_t) { return s_bind_return; }
-int listen (int, int) { return s_listen_return; }
-#else
+#if defined __GLIBC__ && !defined __FreeBSD__
```

Comment:
Looks good, just would change this:
```suggestion
#ifde __GLIBC__
```

Reply:
Accepted. I just did not want to change the existing behavior. I'm not a BSD guy at all, so I don't pretend to know the implications of the change.